frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -32,26 +47,42 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
         apiVersion: 'v1',
         propagationPolicy: 'Foreground',
       };
-      handlePromise(
-        Promise.all([
-          k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
-          ...(subscription?.status?.installedCSV
-            ? [
-                k8sKill(
-                  ClusterServiceVersionModel,
-                  {
-                    metadata: {
-                      name: subscription.status.installedCSV,
-                      namespace: subscription.metadata.namespace,
-                    },
+      const k8kills = [
+        k8sKill(SubscriptionModel, subscription, {}, deleteOptions),
+        ...(subscription?.status?.installedCSV
+          ? [
+              k8sKill(
+                ClusterServiceVersionModel,
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
                   },
-                  {},
-                  deleteOptions,
-                ).catch(() => Promise.resolve()),
-              ]
-            : []),
-        ]),
-      )
+                },
+                {},
+                deleteOptions,
+              ).catch(() => Promise.resolve()),
+            ]
+          : []),
+      ];
+      if (deleteAllOperands && operands) {
+        operands.forEach((operand: K8sResourceKind) => {
+          const model =
+            models?.[
+              operand.apiVersion
+                .split('/')
+                .join('~')
+                .concat(`~${operand.kind}`)
+            ];
+          if (model) {
+            k8kills.push(k8sKill(model, operand, {}, deleteOptions));
+          } else {
+            // eslint-disable-next-line no-console
+            console.error(`Could not delete ${operand?.metadata?.name}.  Could not find K8s Model`);
```

Comment:
This should not be just a console.error... We have to surface an error to the user if we can't delete something.

Reply:
I noticed that we do not do error checking for `modelFor(referenceFor(<resource>));` anywhere else in the codebase.  I was purposefully creating a false `name` in `<resource>` for testing, but I don't think that's realistic.